python/paddle/fluid/layers/distributions.py

```diff
@@ -0,0 +1,380 @@
+            else:
+                return output
+
+    def log_prob(self, value):
+        """Log probability density/mass function.
+
+        Args:
+          value (Variable): The input tensor.
+
+        Returns:
+          Variable: log probability.
+
+        """
+        lb_bool = control_flow.less_than(self.low, value)
+        ub_bool = control_flow.less_than(value, self.high)
+        lb = tensor.cast(lb_bool, dtype='float32')
+        ub = tensor.cast(ub_bool, dtype='float32')
+        return nn.log(lb * ub) - nn.log(self.high - self.low)
+
+    def entropy(self):
+        """Shannon entropy in nats.
+
+        Returns:
+          Variable: Shannon entropy of uniform distribution.
+
+        """
+        return nn.log(self.high - self.low)
+
+
+class Normal(Distribution):
+    """The Normal distribution with location `loc` and `scale` parameters.
+
+    Mathematical details
+
+    The probability density function (pdf) is,
+
+    .. math::
+
+        pdf(x; \mu, \sigma) = \\frac{1}{Z}e^{\\frac {-0.5 (x - \mu)^2}  {\sigma^2} }
+
+    .. math::
+
+        Z = (2 \pi \sigma^2)^{0.5}
+
+    In the above equation:
+
+    * :math:`loc = \mu`: is the mean.
+    * :math:`scale = \sigma`: is the std.
+    * :math:`Z`: is the normalization constant.
+    
+    Examples:
+        .. code-block:: python
+
+          from paddle.fluid import layers
+          from paddle.fluid.layers import Normal
+
+          # Define a single scalar Normal distribution.
+          dist = Normal(loc=0., scale=3.)
+          # Define a batch of two scalar valued Normals.
+          # The first has mean 1 and standard deviation 11, the second 2 and 22.
+          dist = Normal(loc=[1, 2.], scale=[11, 22.])
+          # Get 3 samples, returning a 3 x 2 tensor.
+          dist.sample([3])
+
+          # Define a batch of two scalar valued Normals.
+          # Both have mean 1, but different standard deviations.
+          dist = Normal(loc=1., scale=[11, 22.])
+
+          # Define a batch of two scalar valued Normals.
+          # Both have mean 1, but different standard deviations.
+          dist = Normal(loc=1., scale=[11, 22.])
+
+          # Variable as input
+          dims = 3
+
+          loc = layers.data(name='loc', shape=[dims], dtype='float32')
+          scale = layers.data(name='scale', shape=[dims], dtype='float32')
+          other_loc = layers.data(
+              name='other_loc', shape=[dims], dtype='float32')
+          other_scale = layers.data(
+              name='other_scale', shape=[dims], dtype='float32')
+          values = layers.data(name='values', shape=[dims], dtype='float32')
+
+          normal = Normal(loc, scale)
+          other_normal = Normal(other_loc, other_scale)
+
+          sample = normal.sample([2, 3])
+          entropy = normal.entropy()
+          lp = normal.log_prob(values)
+          kl = normal.kl_divergence(other_normal)
+    """
+
+    def __init__(self, loc, scale):
```

Comment:
Whether need dtype as the input, since hard code float32 is used in many places.